src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipePayloadDecoder.cs

```diff
@@ -128,49 +115,26 @@ internal static object[] DecodePayload(ref EventSource.EventMetadata metadata, R
             return decodedFields;
         }
 
-        private static UInt64 ReadUnalignedUInt64(ref ReadOnlySpan<byte> payload)
-        {
-            UInt64 val = 0;
-            if (BitConverter.IsLittleEndian)
-            {
-                val |= MemoryMarshal.Read<UInt32>(payload);
-                payload = payload.Slice(sizeof(UInt32));
-                val |= (MemoryMarshal.Read<UInt32>(payload) << sizeof(UInt32));
-                payload = payload.Slice(sizeof(UInt32));
-            }
-            else
-            {
-                val |= (MemoryMarshal.Read<UInt32>(payload) << sizeof(UInt32));
-                payload = payload.Slice(sizeof(UInt32));
-                val |= MemoryMarshal.Read<UInt32>(payload);
-                payload = payload.Slice(sizeof(UInt32));
-            }
-
-            return val;
-        }
-
         private static Guid ReadUnalignedGuid(ref ReadOnlySpan<byte> payload)
         {
             const int sizeOfGuid = 16;
-            byte[] guidBytes = new byte[sizeOfGuid];
+            Guid guid;
             if (BitConverter.IsLittleEndian)
             {
+                Span<byte> guidBytes = stackalloc byte[sizeOfGuid];
                 for (int i = sizeOfGuid - 1; i >= 0; i--)
```

Comment:
This reversing logic looks quite weird - why would the GUID be reversed on little-endian systems?

Reply:
Agree, The Guid constructor deals with endianess already. This little endian paths looks broken and unnecessary.

Reply:
Removed it.